vnext/Shared/Networking/WinRTWebSocketResource.cpp

```diff
@@ -382,7 +382,10 @@ void WinRTWebSocketResource::Connect(string &&url, const Protocols &protocols, c
         scheme = L"https";
       }
 
-      auto origin = winrt::hstring{scheme + L"://" + host + L":" + winrt::to_hstring(port)};
+      // Only add a port if a port is defined
+      winrt::hstring originPort = port != 0 ? L":" + winrt::to_hstring(port) : L"";
```

Comment:
For follow-up:

While this change prevents from appending a port with value `0` when the intended value may have been the protocol defaults (`80`, `443`), this would prevent a client app from explicitly using such port, even though it's against conventions and possibly standards.

We should use the same URL processing as [OriginPolicyHttpFilter.cpp](https://github.com/microsoft/react-native-windows/pull/12791/files#diff-9c79fff824b20fbf2186f3381a9bd5b6ba52ed49b87b58ca9913959a68cd952b) to prevent a couple corner cases such as explicitly set default ports.\
See #12626 and https://github.com/OfficeDev/office-js/issues/4144.